### ISSUE TITLE
Command line option to have Language Server inherit VS Code's http agent

### ISSUE THREAD

**Issue by isc-bsaviano** [microsoft/vscode#145559]
I maintain a Language Server written in TypeScript that uses frequent HTTP(S) requests to resolve information from an external server. VS Code helpfully loads the host OS's CA cert store on startup. Is there an option to pass this info along to Language Server process when VS Code starts it up? If not, is there any documentation on how to use the `vscode-proxy-agent` npm module to replicate VS Code's behavior in an extension?

**Comment by chrmarti** [comment]
Not at the moment. `vscode-proxy-agent` also needs a way to look up the proxy from the OS and currently does that through Electron on which VS Code runs.

I think the JSON language server sends its HTTP requests through the JSON extension, maybe that is a workaround you could use too? /cc @aeschli

**Comment by aeschli** [comment]
Yes, the JSON language server forwards the requests to the client running in the extension host.
